File: ios/ClearHiiT/WorkoutSyncModule.m
#import <React/RCTBridgeModule.h>

@interface RCT_EXTERN_MODULE(WorkoutSync, NSObject)

RCT_EXTERN_METHOD(syncSessionsData:(NSString *)json)

RCT_EXTERN_METHOD(syncPreferences:(BOOL)hideFolders)

@end

File: ios/ClearHiiT/WorkoutSyncModule.swift
import Foundation

@objc(WorkoutSync)
class WorkoutSync: NSObject {

  @objc
  func syncSessionsData(_ json: String) {
    WorkoutStore.shared.applySessionsDataJSON(json)
  }

  @objc
  func syncPreferences(_ hideFolders: Bool) {
    WorkoutStore.shared.applyPreferences(hideFolders: hideFolders)
  }

  @objc
  static func requiresMainQueueSetup() -> Bool {
    return false
  }
}
